components/support/error/macros/src/argument.rs

```diff
@@ -2,16 +2,33 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-pub(crate) fn validate(arguments: &syn::AttributeArgs) -> syn::Result<()> {
-    // For now we validate that there are no arguments.
-    // if we want to improve the macro to support attributes, for example,
-    // if we'd like to pass in the return type of the body we can modify this
-    // logic to parse and return the type.
-    if !arguments.is_empty() {
-        return Err(syn::Error::new(
+/// Returns the name of the type of the "internal" error.
+pub(crate) fn validate(arguments: &syn::AttributeArgs) -> syn::Result<String> {
+    // We insist on a single arg of form `E=ErrorType`. If we end up growing
+    // more args we'd probably just add a struct and return that, but this seems
+    // ok for now.
+    fn oops(inner: &str) -> syn::Error {
+        syn::Error::new(
             proc_macro2::Span::call_site(),
-            "Expected #[handle_error] with no arguments",
-        ));
+            format!("Expected #[handle_error({inner})]"),
+        )
+    }
+    if arguments.len() != 1 {
+        return Err(oops(".."));
+    }
+
+    let nested_meta = arguments.iter().next().unwrap();
+    match nested_meta {
+        syn::NestedMeta::Meta(meta) => match meta {
+            syn::Meta::NameValue(meta_name_value) => match meta_name_value.path.get_ident() {
```

Comment:
What if we use a `Meta::Path` instead of a `NameValue`

We'd lose the `E = ` but we can have the macro take the type itself.

so `#[handler_error(Error)]` would be the way to use it. I don't feel strongly, but it's a little nicer not having to put the type in a string literal :(

Reply:
if we do ^ we'd have to return the path or a `TokenStream` from the `validate` function, which seems OK?